lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/REEFLauncher.java

```diff
@@ -204,21 +175,12 @@ public static void main(final String[] args) {
 
     final REEFLauncher launcher = getREEFLauncher(args[0]);
 
-    Thread.setDefaultUncaughtExceptionHandler(new REEFUncaughtExceptionHandler(launcher.clockConfig));
-    launcher.reefVersion.logVersion(); // Write REEF version to the log.
-
-    try (final Clock clock = launcher.getClockFromConfig()) {
-
-      LOG.log(Level.FINE, "Clock: start");
-      clock.run();
-      LOG.log(Level.FINE, "Clock: exit normally");
+    Thread.setDefaultUncaughtExceptionHandler(new REEFUncaughtExceptionHandler(launcher.envConfig));
 
+    try (final REEFEnvironment reef = new REEFEnvironment(launcher.envConfig)) {
```

Comment:
That's a chicken and egg problem. By making the whole `REEFEnvironment` injectable, we lose control over the order of object instantiation, and here we need to create an error handler before we create a `Clock` object. Otherwise it would be impossible e.g. to handle errors in the driver constructor properly.